### ISSUE TITLE
Restoring HAOS backup on Yellow wiped the attached network drives over SAMBA.

### ISSUE THREAD

**Issue by Siress** [home-assistant/supervisor#4726]
### The problem

TL;DR - or how to replicate: Using HA Yellow with OS v. 2023.11.3. Connect a network drive over SAMBA title "media". Create backups using HA Google Drive Backup v0.111.1 add-on. Restore the locally-generated backup using HAOS. Watch contents of 'media' network drive vanish.

Can anyone replicate this? I'm going off of log files and correlation with n=1.

It took me quite a while to piece together why my Media folder went from 8TB to a few empty folders. Checking the ZFS logs gave me the timestamp that the change happened. My NAS has been a rock solid appliance and nothing else was out of the ordinary, so I was sure it had to be something with credentials to execute a deletion. That helped limit it to just a few devices. Foolishly, I had given that privilege to HAOS. Originally, I was giving it access to a share to store my HAOS backups and database. In all of my excitement for HAOS's capabilities, I foolishly added my Media share just to experiment with HAOS's media features.

Given the timestamp and the limited devices with the authority to make the change, the only devices that were active at that time were my HA Yellow and my workstation - as I was integrating various things into HA. After one of the integrations went sideways, I decided to restore a backup. I created a backup just before restoring, just in case. I'm using the Home Assistant Google Drive Backup (0.111.1) add-on to create backups, so they go to the cloud as well as my NAS. Keep in mind that it does not restore backups. I then went through HAOS and restored the backup from the NAS. The time stamp on that backup coincided with the time stamp in the ZFS logs. I then realized that it was authorized to delete the Backups dataset as well, but it hadn't deleted it's own backups. Maybe that's useful knowledge for whomever can fix this issue.

Shame on me for granting excess privilege to HAOS. I take full responsibility for that part of this story. I love HAOS, but this is one of the worst bugs I've come across. This needs immediate action.

### What version of Home Assistant Core has the issue?

core-2023.11.3

### What was the last working version of Home Assistant Core?

core-2023.11.3

### What type of installation are you running?

Home Assistant OS

**Comment by agners** [comment]
This very much sounds like #4652, however we added additional safeguards so that would not happen anymore. I try to reproduce your issue on my end again.

**Comment by agners** [comment]
> Shame on me for granting excess privilege to HAOS. I take full responsibility for that part of this story. I love HAOS, but this is one of the worst bugs I've come across. This needs immediate action.

I am very sorry what happened, and I agree this is a really bad bug!

We had two safe guards in place, unmounting the share and using rm to not cross file system boundaries. Unfortunately, both safeguards failed in their own way. The first one was a race condition, so that might depend on the hardware/system if it actually appears. But I was able to reproduce on my end. Thanks for your detailed report!

With #4732 in merged, we have at least one working mitigation in place. Supervisor 2023.11.6 now on the stable channel contains that fix.
